cub/test/thread_reduce/catch2_test_thread_reduce_partial.cu

```diff
@@ -0,0 +1,611 @@
+__global__ void
+thread_reduce_partial_kernel_span(const T* d_in, T* d_out, ReduceOperator reduce_operator, int valid_items)
+{
+  T thread_data[NUM_ITEMS];
+
+  _CCCL_PRAGMA_UNROLL_FULL()
+  for (int i = 0; i < NUM_ITEMS; ++i)
+  {
+    thread_data[i] = d_in[i];
+  }
+  ::cuda::std::span<T, NUM_ITEMS> span(thread_data);
+  *d_out = cub::detail::ThreadReducePartial(span, reduce_operator, valid_items);
+}
+
+#if _CCCL_STD_VER >= 2023
+
+template <int NUM_ITEMS, typename T, typename ReduceOperator>
+__global__ void
+thread_reduce_partial_kernel_mdspan(const T* d_in, T* d_out, ReduceOperator reduce_operator, int valid_items)
+{
+  T thread_data[NUM_ITEMS];
+
+  _CCCL_PRAGMA_UNROLL_FULL()
+  for (int i = 0; i < NUM_ITEMS; ++i)
+  {
+    thread_data[i] = d_in[i];
+  }
+  using Extent = ::cuda::std::extents<int, NUM_ITEMS>;
+  ::cuda::std::mdspan<T, Extent> mdspan(thread_data, ::cuda::std::extents<int, NUM_ITEMS>{});
+  *d_out = cub::detail::ThreadReducePartial(mdspan, reduce_operator, valid_items);
+}
+
+#endif // _CCCL_STD_VER >= 2023
+
+/***********************************************************************************************************************
+ * CUB operator to STD operator
+ **********************************************************************************************************************/
+
+template <typename T, typename>
+struct cub_operator_to_std;
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::plus<>>
+{
+  using type = ::std::plus<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::multiplies<>>
+{
+  using type = ::std::multiplies<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::bit_and<>>
+{
+  using type = ::std::bit_and<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::bit_or<>>
+{
+  using type = ::std::bit_or<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::std::bit_xor<>>
+{
+  using type = ::std::bit_xor<T>; // T: MSVC complains about possible loss of data
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::minimum<>>
+{
+  using type = ::cuda::minimum<>;
+};
+
+template <typename T>
+struct cub_operator_to_std<T, ::cuda::maximum<>>
+{
+  using type = ::cuda::maximum<>;
+};
+
+template <typename T, typename Operator>
+using cub_operator_to_std_t = typename cub_operator_to_std<T, Operator>::type;
+
+/***********************************************************************************************************************
+ * CUB operator to identity
+ **********************************************************************************************************************/
+
+template <typename T, typename Operator, typename = void>
+struct cub_operator_to_identity;
```

Comment:
Still need to note that `identity_v` should be used once https://github.com/NVIDIA/cccl/pull/4312 is merged.

Reply:
Already done in #5300. Waiting for it to get merged?